lib/rucio/daemons/reaper/reaper2.py

```diff
@@ -240,6 +240,13 @@ def get_rses_to_hostname_mapping():
 
     result = REGION.get('rse_hostname_mapping')
     if result is NO_VALUE:
+        semaphore = REGION.get('semaphore')
```

Comment:
i don't think this is correct. multiple reapers on the same node could set/delete this value. the REGION.get() needs to atomically wait, not check it with an `if` later. or am i misunderstanding what you're trying to accomplish here?